src/net/p2p/nodeconns.cpp

```diff
@@ -65,6 +73,22 @@ std::unordered_map<P2P::NodeID, P2P::NodeInfo, SafeHash> P2P::NodeConns::getConn
   return this->nodeInfo_;
 }
 
+std::unordered_map<P2P::NodeID, P2P::NodeType, SafeHash> P2P::NodeConns::getConnectedWithNodeType() {
+  std::unordered_map<NodeID, NodeType, SafeHash> nodesToNodeType;
+  std::scoped_lock lock(this->stateMutex_);
+  for (const auto& node : nodeInfo_) {
+    nodesToNodeType[node.first] = P2P::NodeType::NORMAL_NODE; // FIXME/REVIEW: are we dealing here exclusively with full nodes?
```

Comment:
In the last commit I'm just getting the correct node type info from the sessions map. The only reason this is in NodeConns is because NodeInfo now includes a node list, and the node list uses the same serialization code as the previous node list message (`RequestNodes`) whose serialization scheme did include the node type. To  avoid creating a node list without node type, I'm just doing the work of fetching this info from the sessions map and including it. If this turns out to be unnecessary I can remove all of this later.